Enable DROP TABLE statement for Presto

tomz opened the pull request.

tomz committed:
Add T_DropStmt check to prestogres_send_to_where

tomz committed:
Set prestogres_system_catalog_relcache = NULL for DropStmt

tomz committed:
Add comments for resetting prestogres_system_catalog_relcache after drop stmt

frsyuki commented on src/context/pool_query_context.c:
```diff
@@ -1825,7 +1831,7 @@ PRESTOGRES_DEST prestogres_send_to_where(Node *node)
 	 * CREATE TABLE ... AS SELECT
 	 */
-	if (IsA(node, SelectStmt) || IsA(node, InsertStmt) || IsA(node, CreateStmt) || IsA(node, CreateTableAsStmt))
+	if (IsA(node, SelectStmt) || IsA(node, InsertStmt) || IsA(node, CreateStmt) || IsA(node, CreateTableAsStmt) || IsA(node, DropStmt))
```
@tomz  Nice :+1: 
Setting NULL directly prestogres_system_catalog_relcache seems leaking memory. And I found `prestogres_discard_system_catalog` function which is called at https://github.com/tomz/prestogres/blob/master/src/protocol/pool_proto_modules.c#L1638-L1645
So probably the best place to discard system catalog is right after the `if` block as following:

```
 if (IsA(node, CreateStmt))
 {
     CreateStmt *create_table_stmt = (CreateStmt *)node;
     if (create_table_stmt->relation->relpersistence)
         discard_temp_table_relcache();
     /* prestogres: discard system catalog to recreate them next time at prestogres_init_system_catalog */
     prestogres_discard_system_catalog();
 }
+ else if (IsA(node, DropStmt))
+ {
+     /* prestogres: discard system catalog to recreate them next time at prestogres_init_system_catalog */
+     prestogres_discard_system_catalog();
+ }
```

tomz replied:
Ok thanks, will do that instead.

tomz committed:
Move the discard system catalog logic to pool_proto_modules.c

tomz committed:
restrict access to 'system' databases; and also merged in 'Hide temporary created pg_roles by Prestogres #37'

tomz committed:
Fixed typo

tomz committed:
Revert PR #37

tomz committed:
template should be template0

tomz committed:
Put PR #37 back

xerial committed:
Fix when a table contains more than 1600 columns (a limit of postgres catalog)

tomz committed:
Allow passing CREATE VIEW to Presto

tomz committed:
Call prestogres_discard_system_catalog() for ViewStmt

tokaaron committed:
Remove code that improperly calls pool_unread

Presto queries may result in outputs that contain multiple
\0 or NULL characters.  As a result, a single error
message or notice may contain multiple \0 delimited
strings which would get read at once on invocation to
per_node_error_log(), which then attempts to 'unread'
the contents it just read.  However, the pool_unread invocation
ends up unreading up to the first \0 character, and it
does not unread the full message, causing subsequent
calls to getting the postgres message kind to fail
as the kind ends up being a \0 character.

wontonst committed:
Add presto variable SET and SET SESSION SHOW cmds

wontonst committed:
Add SET CHARACTERISTIC and multistatement SET passthrough to psql

This change allows JDBCv2 clients to work by allowing SET TRANSACTION
statements such as SET SESSION CHARACTERISTIC as well as
multistatement queries containing SET to route to psql rather than
Presto.

tomz committed:
Merge pull request #2 from wontonst/tokhy-merge

Merge in SET passthrough to presto, remove code that improperly calls pool_unread

wontonst committed:
Add application_name to list of psql cmds

tomz committed:
Merge pull request #3 from wontonst/app_name

Add application_name to list of psql cmds

wontonst committed:
Fix Presto RESTful API link traverse ending early

tomz committed:
Merge remote-tracking branch 'treasure-data/master'
